subsys/bluetooth/host/hci_core.c

```diff
@@ -4118,6 +4215,25 @@ static int le_init(void)
 	}
 #endif
 
+#if defined(CONFIG_BT_WHITELIST)
+	struct bt_hci_rp_le_read_wl_size *rp;
```

Comment:
Im not a found of declaration in the middle of the function, specially since this is a conditional block, we can probably have a cleaner code by moving this into its own function.

Reply:
I will create an _complete function, which seems to be the pattern.